subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/suite/filter/TestsFilter.kt

```diff
@@ -0,0 +1,54 @@
+package com.avito.instrumentation.suite.filter
+
+import com.avito.instrumentation.suite.dex.AnnotationData
+import com.avito.report.model.DeviceName
+import java.io.Serializable
+
+interface TestsFilter {
+
+    sealed class Result {
+        object Included : Result()
+        abstract class Excluded(val reason: String) : Result() {
+            class HaveSkipSdkAnnotation(sdk: Int) : Excluded("test has SkipSdk with value sdk=$sdk")
+            class DoNotHaveIncludeAnnotations(annotations: Set<String>) : Excluded("test doesn't have any of annotations=$annotations")
+            class HaveExcludeAnnotations(annotations: Set<String>) : Excluded("test has any of excluded annotations=$annotations")
+            abstract class BySignatures(reason: String): Excluded(reason) {
+                abstract val source: Signatures.Source
+            }
+            class DoNotMatchIncludeSignature(
+                override val source: Signatures.Source,
+                signatures: Set<Signatures.TestSignature>
+            ) : BySignatures("test doesn't match any of signatures=$signatures")
+
+            class MatchExcludeSignature(
+                override val source: Signatures.Source,
+                signatures: Set<Signatures.TestSignature>
+            ) : BySignatures("test has matched one of signatures=$signatures")
+        }
+    }
+
+    data class Test(
+        val name: String,
+        val annotations: List<AnnotationData>,
+        val deviceName: DeviceName,
+        val api: Int
+    ) {
+        companion object
+    }
+
+    data class Signatures(
+        val source: Source,
+        val signatures: Filter.Value<TestSignature>
+    ) {
+        enum class Source {
+            Code, ImpactAnalysis, PreviousRun, Report
```

Comment:
As I said at SignaturesFilter it needs to know what type of Signature filter has filtered a test.